indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -146,7 +151,7 @@
   @Nullable
   private final ClientCompactionTaskGranularitySpec granularitySpec;
   @Nullable
-  private final ParallelIndexTuningConfig tuningConfig;
+  private final CompactionTuningConfig tuningConfig;
```

Comment:
I think this change means that compaction jobs that are submitted with a `ParallelIndexTuningConfig` (using type = `index_parallel`) will start failing after this change - is this correct?

If so, I think instead of introducing this as a breaking change, we can just add the Precondition check that you have in the `CompactionTuningConfig`  into the constructor for the compaction task.

If this isn't a breaking change, I like that there's a separate tuningConfig for compaction tasks, so that in the future, this config can be more easily optimized.


cc @maytasm Since I've seen you make some improvements around compaction recently

Reply:
I take this back. `getTuningConfig` deals with the different types of tuningConfigs. 🤦